test/uk/gov/hmrc/securemessage/controllers/SecureMessageControllerSpec.scala

```diff
@@ -208,25 +225,48 @@ class SecureMessageControllerSpec extends PlaySpec with ScalaFutures with Mockit
 
   "createCaseworkerMessage" must {
     "return CREATED (201) when with valid payload" in new CreateCaseWorkerMessageTestCase(
-      requestBody = Resources.readJson("model/api/caseworker-message.json")) {
+      requestBody = Resources.readJson("model/api/caseworker-message.json"),
+      givenResult = Future.successful(Right(()))) {
       private val response = controller.createCaseworkerMessage("cdcm", "123")(fakeRequest)
       status(response) mustBe CREATED
     }
+    "return UNAUTHORIZED (401) when the caseworker is not a conversation participant" in new CreateCaseWorkerMessageTestCase(
+      requestBody = Resources.readJson("model/api/caseworker-message.json"),
+      givenResult = Future.successful(Left(NoCaseworkerIdFound("Caseworker ID not found")))
+    ) {
+      private val response = controller.createCaseworkerMessage("cdcm", "123")(fakeRequest)
+      status(response) mustBe UNAUTHORIZED
+    }
+    "return BAD_REQUEST (400) when the message content is not base64 encoded" in new CreateCaseWorkerMessageTestCase(
+      requestBody = Resources.readJson("model/api/caseworker-message.json"),
+      givenResult = Future.successful(Left(InvalidBase64Content("Not valid base64 content")))
+    ) {
+      private val response = controller.createCaseworkerMessage("cdcm", "123")(fakeRequest)
+      status(response) mustBe BAD_REQUEST
+      contentAsString(response) mustBe "\"Not valid base64 content\""
+    }
+    "return BAD_REQUEST (400) when the message content is not valid HTML" in new CreateCaseWorkerMessageTestCase(
+      requestBody = Resources.readJson("model/api/caseworker-message.json"),
+      givenResult = Future.successful(Left(InvalidHtmlContent("Not valid HTML content")))) {
```

Comment:
i love this, the other two comments about moving mock to context i posted above should to the same